src/Build/Construction/ProjectImportElement.cs

```diff
@@ -38,33 +52,35 @@ internal ProjectImportElement(XmlElementWithLocation xmlElement, ProjectRootElem
         /// </summary>
         public string Project
         {
-            get => FileUtilities.FixFilePath(ProjectXmlUtilities.GetAttributeValue(XmlElement, XMakeAttributes.project));
+            get => FileUtilities.FixFilePath(GetAttributeValue(XMakeAttributes.project));
             set
             {
                 ErrorUtilities.VerifyThrowArgumentLength(value, XMakeAttributes.project);
 
-                ProjectXmlUtilities.SetOrRemoveAttribute(XmlElement, XMakeAttributes.project, value);
-                MarkDirty("Set Import Project {0}", value);
+                SetOrRemoveAttribute(XMakeAttributes.project, value, "Set Import Project {0}", value);
             }
         }
 
         /// <summary>
         /// Location of the project attribute
         /// </summary>
-        public ElementLocation ProjectLocation => XmlElement.GetAttributeLocation(XMakeAttributes.project);
+        public ElementLocation ProjectLocation => GetAttributeLocation(XMakeAttributes.project);
 
         /// <summary>
         /// Gets or sets the SDK that contains the import.
         /// </summary>
         public string Sdk
         {
-            get => FileUtilities.FixFilePath(ProjectXmlUtilities.GetAttributeValue(XmlElement, XMakeAttributes.sdk));
+            get => FileUtilities.FixFilePath(GetAttributeValue(XMakeAttributes.sdk));
             set
             {
                 ErrorUtilities.VerifyThrowArgumentLength(value, XMakeAttributes.sdk);
+                // TODO: here and other setters. Confirm that CheckUpdatedSdk() ussage is
+                //       really the intended one. Currently it does check for equality  before updating the attribute
+                //       so setter will only take effect if "ParsedSdkReference" is not updated. Currently it might work by accident
+                //       if all "Set" calls are different and no ParsedSdkReference is triggered in between.
```

Comment:
Yeah, this looks wrong to me, too. Can you file a bug for us to look at? You're not making this worse, fortunately.

Reply:
opened:
Bug 978973